cloudpickle/cloudpickle.py

```diff
@@ -1142,30 +1136,31 @@ def _rehydrate_skeleton_class(skeleton_class, class_dict):
     return skeleton_class
 
 
-def _can_find_module(mod_name):
+def _is_dynamic(module):
     """
-    Iterate over each part instead of calling imp.find_module directly.
-    This function is able to find submodules (e.g. os.path).
-
-    Return True if the module can be found.
+    Return True if the module is special module that cannot be imported by its
+    name.
     """
-    if hasattr(importlib, 'util'):
-        spec = importlib.util.find_spec(mod_name)
-        return spec is not None
+    # Quick check: module that have __file__ attribute are not dynamic modules.
+    if hasattr(module, '__file__'):
+        return False
+
+    if hasattr(module, '__spec__'):
+        return module.__spec__ is None
```

Comment:
Nice one!! This is indeed better

Reply:
The old imp based code is fishy to me anyway. I cannot find a valid use case where `__file__` would not be defined and the `imp`-based thingy would work.

Reply:
In particular, I cannot build a valid case that would make it possible to have perfect coverage of the `_is_dynamic` function.

I don't think it was covered either before.